packages/grid/x-data-grid/src/constants/envConstants.ts

```diff
@@ -22,13 +22,13 @@ let experimentalEnabled = false;
 
 if (
   typeof process !== 'undefined' &&
-  process.env.GRID_EXPERIMENTAL_ENABLED !== undefined &&
```

Comment:
I haven't used this flag and didn't even know it exists.
We rather use `experimentalFeatures` prop which is usually mentioned in the docs and - see [Editing](https://mui.com/x/react-data-grid/editing/) or [Aggregation](https://mui.com/x/react-data-grid/aggregation/) for example.

There's only one place using this flag:
https://github.com/mui/mui-x/blob/9593984c724e16cf6a06b244c427a7d3b197a564/packages/grid/x-data-grid/src/components/panel/GridColumnsPanel.tsx#L156

You can see it in action in [this PR deployment](https://deploy-preview-4863--material-ui-x.netlify.app/x/react-data-grid/demo/), but those are just disabled buttons that do nothing:

<img width="480" alt="Screenshot 2022-08-01 at 14 24 00" src="https://user-images.githubusercontent.com/13808724/182147258-27a1db30-234e-4540-8c41-934262b3c698.png">

So I'm not sure what was the goal here.

I would rather remove this flag from the codebase completely.
We can keep exporting `GRID_EXPERIMENTAL_ENABLED` set to `false` just for the sake of 100% backward compatibility.

Reply:
Totally agree with your suggestion if we can go this route.
I've also checked it and didn't understand the reason behind it, because even with it enabled the feature does seem disabled.
Maybe @DanailH remembers the reason behind it?

Reply:
I think this was added before because of the documentation - so specific features did not appear there.

Reply:
@LukasTy @DanailH @cherniavskii 

Would you like me to update the PR with - 
```typescript
export const GRID_EXPERIMENTAL_ENABLED = false
```
?

Reply:
To me it would seem like a reasonable fix, at least for now, but I don't know what is/was the teams plan with this unfinished column re-ordering feature, hence whoever worked on it could suggest the best course of action.

Reply:
@ar7casper I think so, unless @DanailH has objections

Reply:
I think we can move forward.

Reply:
@DanailH @cherniavskii Done, can you allow the pipeline please?

Reply:
Should we maybe also remove the logic related to populating `GRID_EXPERIMENTAL_ENABLED` environment variable as well?
There are two instances. One in `docs/next.config.js` and other in `packages/storybook/.storybook/main.ts`

Reply:
I've updated the PR to remove unnecessary occurrences of `GRID_EXPERIMENTAL_ENABLED`